### DISCUSSION TITLE
Spatie setting plugin

### DISCUSSION THREAD

**Discussion by anishsarkar** [filamentphp/filament#1399] [Help]
How does one use the lock facility provided by the spatie package? The plugin seems to ignore whether the setting in question is locked or not. 

Is there a way to view the settings in a table form, with the information such as locked and a link to open a modal which will allow to update as will as lock/unlock the setting?

**Comment by MACscr** [comment]
What do you mean by locked/unlocked? This is just a simple wrapper that creates a nav item and some basic save functionality. Everything else is pure spatie and you would have to learn more about its package to use its functionality.

**Reply by anishsarkar** [reply]
The spatie package provides a facility for setting to be **locked / unlocked**. If a setting is locked it can't be updated. 

This wrapper does not check whether a setting is locked or not. So, I was wondering if there is a way to show whether the setting is locked or not, if locked unlock it, if unlocked lock it and provide the update facility only if the setting is unlocked. 

 I was thinking that I will make a table which shows the locked status and provide action links for locking / unlocking and editing when unlocked which will open a modal form for updating. But the problem is that these are not eloquent models, and I have not dug deeper to see how to make tables out of collections.
